hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/WriteSocketTextStreamP.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.impl.connector;
+
+import com.hazelcast.jet.AbstractProcessor;
+import com.hazelcast.jet.Processor;
+import com.hazelcast.jet.ProcessorSupplier;
+
+import javax.annotation.Nonnull;
+import java.io.BufferedWriter;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.net.Socket;
+import java.util.List;
+
+import static com.hazelcast.jet.impl.util.Util.uncheckRun;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.IntStream.range;
+
+/**
+ * A writer that connects to a specified socket and writes the items as text.
+ * This processor expects a server-side socket to be available for connecting to
+ * <p>
+ * Each processor instance will create a socket connection to the configured [host:port]
+ */
+public class WriteSocketTextStreamP extends AbstractProcessor implements Closeable {
+
+    private final String host;
+    private final int port;
+    private Socket socket;
+    private BufferedWriter bufferedWriter;
+
+    WriteSocketTextStreamP(String host, int port) {
+        this.host = host;
+        this.port = port;
+    }
+
+    /**
+     * Creates a supplier for {@link WriteSocketTextStreamP}
+     *
+     * @param host The host name to connect to
+     * @param port The port number to connect to
+     */
+    public static ProcessorSupplier supplier(String host, int port) {
+        return new Supplier(host, port);
+    }
+
+    @Override
+    protected void init(@Nonnull Context context) throws Exception {
+        getLogger().info("Connecting to socket " + hostAndPort());
+        socket = new Socket(host, port);
+        getLogger().info("Connected to socket " + hostAndPort());
+        bufferedWriter = new BufferedWriter(new OutputStreamWriter(socket.getOutputStream(), "UTF-8"));
+    }
+
+    @Override
+    protected boolean tryProcess(int ordinal, @Nonnull Object item) throws Exception {
+        bufferedWriter.write(item.toString());
+//        bufferedWriter.flush();
```

Comment:
commented out line.

This class should implement `Processor` directly and flush after consuming all of the inbox.